File: JavaJungsuk/ExampleCode/ch5/BubbleSort.java
package ExampleCode.ch5;

public class BubbleSort {
    public static void main(String[] args) {
        int[] arr = new int[10];                        //배열 생성

        for(int i=0; i< arr.length; i++){               //생성된 배열요소에 0~9 정수 넣기
            arr[i] = (int)(Math.random()*10);
            System.out.print(arr[i]);
        }
        System.out.println();

        for(int i=0; i<arr.length-1; i++){
            boolean changed = false;    //자리바꿈 됐는지 체크

            for(int j=0; j<arr.length-1-i; j++){
                if( arr[j]> arr[j+1]) {             //우측값이 더 작으면 자리교체
                    int tmp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1]= tmp;
                    changed = true;
                }
            }//end for j
            if (!changed) {         //자리교체 없으면 반복문 탈출
                break;          
            }
            for(int k=0; k<arr.length; k++){
                System.out.print(arr[k]);
            }
            System.out.println();
        } //end for i
    }
}

File: python/sort/bubbleSort.py
# 버블 정렬
# 서로 인접한 두 원소를 비교

array = [1, 10, 5, 8, 7, 6, 4, 3, 2, 9]

# 0부터 데이터 길이만큼 반복
for i in range(len(array)):
    # 인접한 두 원소의 비교이므로 j와 j-1끼리의 비교를 위해 1부터 시작
    for j in range(1, len(array)-i):
        # j-1위치의 값이 더 크면 위치이동
        if array[j-1] > array[j]:
            tmp = array[j-1]
            array[j-1] = array[j]
            array[j] = tmp

print(array)
